1. Challenge 1:
  - Answer: b
  - Explanation: estamos dejando el campo del parametro vacio por lo cual cuando llamamos de nuevo la variable foo por fuera nos esta deviolviando lo que le hemos asignado dentro de la funcion.


2. Challenge 2:
  - Answer: c
  - Explanation: Estamos ejecutando console log dentro de la funcion en donde ya hemos declarado que a sera igual a 10, y luego hacemos nuevamente la llamada fuera de la funcion en la que habiamos dicho anteriormente que a es igual a 1, asi que en cuando salga a la consola saldra la respuesta c.


3. Challenge 3:
  - Answer: c
  - Explanation: estamos creando una funcion llamada sayHi luego llamamos a la funcion con un texto, el cual aparece en la consola cuando llamamos a la funcion.


4. Challenge 4:
  - Answer:c
  - Explanation: cuando cambiamos ele valor de la clave "num" se hace tambien en el object original por eso la respuesta es la c


5. Bonus - Challenge 5:
  - Answer:
  - Explanation:
